pyrobosim/pyrobosim/navigation/astar_grid.py

```diff
@@ -0,0 +1,329 @@
+        :type resolution: float
+        :param inflation_radius: The inflation radius to be used for the planner's occupancy grid.
+        :type inflation_radius: float
+        :param distance_metric: The metric to be used as heuristic.
+        :type distance_metric: string 'Manhattan' or 'Euclidean'
+        :param diagonal_motion: If to expand nodes using diagonal motion.
+        :type diagonal_motion: bool
+        :param max_time: Maximum time allowed for planning, in seconds.
+        :type max_time: float
+        """
+
+        self.goal = None
+        self.start = None
+        self.world = world
+        self.visited = set()  # (x, y)
+        self.planning_time = 0
+        self.max_time = max_time
+        self.num_nodes_expanded = 0
+        self.resolution = resolution
+        self.distance_metric = distance_metric
+        self.diagonal_motion = diagonal_motion
+        self.inflation_radius = inflation_radius
+        self.candidates = PriorityQueue()  # Node
+
+        self._set_actions()
+        self._set_heuristic()
+        self._set_occupancy_grid()
+
+    def _set_occupancy_grid(self):
+        """Generates occupancy grid of specified configuration"""
+        print(
+            f"Generating occupancy grid at resolution {self.resolution} and inflation radius {self.inflation_radius}"
+        )
+        ts = time.time()
+        self.grid = occupancy_grid_from_world(
+            self.world,
+            resolution=self.resolution,
+            inflation_radius=self.inflation_radius,
+        )
+        print(f"Generated in : {time.time() - ts} seconds.")
+
+    def _set_actions(self):
+        """Generates the actions available"""
+        self.actions = {
+            "left": {"cost": 1, "action": (-1, 0)},
+            "right": {"cost": 1, "action": (1, 0)},
+            "up": {"cost": 1, "action": (0, 1)},
+            "down": {"cost": 1, "action": (0, -1)},
+            "left_up": {"cost": 1.414, "action": (-1, 1)},
+            "left_down": {"cost": 1.414, "action": (-1, -1)},
+            "right_up": {"cost": 1.414, "action": (1, 1)},
+            "right_down": {"cost": 1.414, "action": (1, -1)},
+        }
+        keys = list(self.actions.keys())
+        self.selected_actions = keys if self.diagonal_motion else keys[:4]
+        print(f"Selected actions : {self.selected_actions}")
+
+    def _set_heuristic(self):
+        """Sets the heuristic"""
+        euclidean = lambda p1, p2: math.sqrt(
+            (p1[0] - p2[0]) ** 2 + (p1[1] - p2[1]) ** 2
+        )
+        manhattan = lambda p1, p2: abs(p1[0] - p2[0]) + abs(p1[1] - p2[1])
+
+        if self.distance_metric == "Euclidean":
+            self._heuristic = euclidean
+        elif self.distance_metric == "Manhattan":
+            if not self.diagonal_motion:
+                warnings.warn("Manhattan over estimates without diagonal motion")
+            self._heuristic = manhattan
+        print(f"Selected heuristic: {self.distance_metric}")
```

Comment:
Should there be an "else" condition? Should either error or print a warning that it fell back to some default

Reply:
In this case, I was not sure if we should override the choice user made .
So I decided to go with warning while keeping the user's choice.